### ISSUE TITLE
Default routes for serviceWorker option

### ISSUE THREAD

**Issue by cyrilletuzi** [angular/angular-cli#5148]
`@angular/cli` rc.0 with `--ng4` option

Currently, the `serviceWorker` option only registers the assets to cache in `ngsw-manifest.json`, but no routing.

So it only works if I go on directly on `http://localhost:4200/index.html`, but not on `http://localhost:4200/` (which could be the url registered in browser favorites for example).

It would be easy to add to the manifest the required options : 
```
"routing": {
  "routes": {
     "/": { "prefix": false },
  },
  "index": "/index.html"
}
```

Not sure if a sub-directory base href would require a different config.

Less trivial, if the app root url is redirecting to another route (let's say "profile"), the manifest should add it too : 
```
"routing": {
  "routes": {
     "/": { "prefix": false },
     "/profile": { "prefix": false },
  },
  "index": "/index.html"
}
```

**Comment by alxhub** [comment]
Hello @cyrilletuzi!

You are correct that your proposed configuration would be a sensible default. Currently it's a little tricky, as the CLI doesn't generate an `ngsw-manifest.json` for you. The integration currently is limited to including hashes of compilation assets seen by Webpack into the manifest.

Longer term, we would like to avoid the user writing this config entirely. The list of routes is statically known to the app via the routing config in `NgModule`s already, we should just leverage this.

Shorter term, if you include either of the `ngsw-manifest.json` files in the CLI project root directory, `ng build --prod` will augment them with the static caching information and output the merged version to `dist`.

**Comment by cyrilletuzi** [comment]
Thanks for the merging tip, it works well.

I'm up to do the missing wiki page about this.

**Comment by mattma** [comment]
@alxhub First of all, thank you for working on Service Worker feature. It is awesome.

I have got the app fully working with 

```
"@angular/core": "4.3.0-rc.0",
...
"@angular/cli": "1.3.0-beta.0",
 "@angular/service-worker": "^1.0.0-beta.16",
```

I am using all the latest version of Angular and Angular CLI.

The question that I have is what option do I use to turn on the **no-cors mode**?

My service worker works pretty fine in the NO-Authentication mode. Once I put it behind an authentication layer, it does not work. Nothing is store in the cache.

To be exact, the issue in the Network tab in Chrome debugger

```
Request URL: `SITE_URL/ngsw-manifest.json?ngsw-cache-bust=0.23123123123`  
Request Method: GET
Status Code: 401 Unauthorized
```

I found this [post](https://stackoverflow.com/questions/41092697/service-worker-fetch-with-authentication), i believe it is what I need.

Note: In the `index.html`, it also run into the same issue when loading `manifest.json` file with `401 Unauthorized`. I used this trick to bypass

`<link rel="manifest" crossOrigin="use-credentials" href="manifest.json">`

But for `ngsw-manifest.json` case, i believe it is a simple configuration option fix. If I am wrong, could you point me to the right place?

**Comment by cyrilletuzi** [comment]
Fixed in service worker final (5.0).
